base_layer/core/src/blocks/genesis_block.rs

```diff
@@ -758,24 +742,15 @@ mod test {
                     output.minimum_value_promise,
                 );
                 let e = PrivateKey::from_bytes(&e_bytes).unwrap();
+
+                // Check that `u_a = e*v`
+                // This _must_ be in addition to a separate check that the signature is valid!
+                // This check is not done here!
                 let value_as_private_key = PrivateKey::from(output.minimum_value_promise.as_u64());
                 assert_eq!(
                     output.metadata_signature.u_a().to_hex(),
-                    (commit_nonce_a.clone() + e.clone() * value_as_private_key).to_hex()
+                    (e.clone() * value_as_private_key).to_hex()
                 );
-
-                // Demonstrating how to extract the value (Note: For the proof, inverting the scalar with multiplication
-                // is less efficient than multiplication only as above)
-                let metadata_signature_u_a_scalar =
-                    Scalar::from_bits(<[u8; 32]>::try_from(output.metadata_signature.u_a().as_bytes()).unwrap());
-                let commit_nonce_a_scalar = Scalar::from_bits(<[u8; 32]>::try_from(commit_nonce_a.as_bytes()).unwrap());
-                let e_scalar = Scalar::from_bits(<[u8; 32]>::try_from(e.as_bytes()).unwrap());
-                let value_scalar = (metadata_signature_u_a_scalar - commit_nonce_a_scalar) * e_scalar.invert();
-                assert_eq!(value_scalar, Scalar::from(output.minimum_value_promise.as_u64()));
-                let mut value_bytes = [0u8; 8];
-                value_bytes.copy_from_slice(&value_scalar.to_bytes()[0..8]);
-                let value = u64::from_le_bytes(value_bytes);
-                assert_eq!(value, output.minimum_value_promise.as_u64());
```

Comment:
I think this is still valid, we just need to remove `commit_nonce_a` from the equations

Reply:
Would it actually be useful? To extract the value, you need the challenge, which requires knowledge of the value to compute.

Reply:
Isn't that somewhat circular? The risk is that if you don't compute the challenge yourself, the proof could be invalid (including a falsified minimum value promise) and you could be fooled into accepting an incorrect value. And if you do compute the challenge yourself, the existing multiplicative assertion is sufficient to show the claimed minimum value promise is valid (if the proof itself is also valid).

Reply:
Yip, agree this does not add value

Reply:
Pun intended, I hope!